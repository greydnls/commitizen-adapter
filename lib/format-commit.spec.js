'use strict'

const { formatCommit } = require('./format-commit')

// Commit details: body, breakingNotes, issues, releaseNotes, subject, type

describe('Commit details formatting', () => {
  test('When there is a type and subject, then they are formatted as the commit header', () => {
    expect(
      formatCommit({
        type: 'New',
        subject: 'Test commit formatting',
      }),
    ).toMatchSnapshot()
  })

  test('When the header is longer than 100 chars, then it is truncated', () => {
    expect(
      formatCommit({
        type: 'New',
        subject:
          'Test commit formatting a really long commit that is longer than 100 characters so it should get sliced',
      }),
    ).toMatchSnapshot()
  })

  test('When a commit emoji is selected, then it is appended to the header', () => {
    expect(
      formatCommit({
        type: 'New',
        subject: 'Test commit emoji',
        emoji: 'ðŸ’–',
      }),
    ).toMatchSnapshot()
  })

  test('When the commit body is longer than 100 chars, then it is wrapped', () => {
    expect(
      formatCommit({
        type: 'New',
        subject: 'Test commit formatting',
        body:
          'This is a longer description of the commit, its over 100 chars so it should be wrapped onto multiple lines but not truncated',
      }),
    ).toMatchSnapshot()
  })

  test('When the commit has breaking changes, then it is appended to the footer', () => {
    expect(
      formatCommit({
        type: 'New',
        subject: 'Test commit emoji',
        breakingNotes: 'This commit includes breaking changes!',
      }),
    ).toMatchSnapshot()
  })

  test('When the commit has release notes, then it is appended to the footer', () => {
    expect(
      formatCommit({
        type: 'New',
        subject: 'Test commit emoji',
        releaseNotes: 'We have worked on this release a long time!',
      }),
    ).toMatchSnapshot()
  })

  test('When the commit closes issues, then it is appended in the footer', () => {
    expect(
      formatCommit({
        type: 'New',
        subject: 'Test commit emoji',
        issues: 'crystal-ball/prettier#47, #98',
      }),
    ).toMatchSnapshot()
  })

  test('Every feature works perfectly ðŸ˜…', () => {
    expect(
      formatCommit({
        type: 'New',
        subject: 'Mega commit',
        emoji: 'ðŸ’–',
        breakingNotes: 'This part of the package has to be updated',
        releaseNotes: 'Its been a long journey, but worth it',
        issues: 'crystal-ball/prettier#47, #98',
      }),
    ).toMatchSnapshot()
  })
})
