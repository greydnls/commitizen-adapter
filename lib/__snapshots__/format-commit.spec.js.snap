// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Commit details formatting Every feature works perfectly ðŸ˜… 1`] = `
"New: Mega commit ðŸ’–

BREAKING CHANGE: This part of the package has to be updated

RELEASE NOTES: Its been a long journey, but worth it

Closes crystal-ball/prettier#47, Closes #98"
`;

exports[`Commit details formatting When a commit emoji is selected, then it is appended to the header 1`] = `"New: Test commit emoji ðŸ’–"`;

exports[`Commit details formatting When the commit body is longer than 100 chars, then it is wrapped 1`] = `
"New: Test commit formatting

This is a longer description of the commit, its over 100 chars so it should be wrapped onto multiple
lines but not truncated"
`;

exports[`Commit details formatting When the commit closes issues, then it is appended in the footer 1`] = `
"New: Test commit emoji

Closes crystal-ball/prettier#47, Closes #98"
`;

exports[`Commit details formatting When the commit has breaking changes, then it is appended to the footer 1`] = `
"New: Test commit emoji

BREAKING CHANGE: This commit includes breaking changes!"
`;

exports[`Commit details formatting When the commit has release notes, then it is appended to the footer 1`] = `
"New: Test commit emoji

RELEASE NOTES: We have worked on this release a long time!"
`;

exports[`Commit details formatting When the header is longer than 100 chars, then it is truncated 1`] = `"New: Test commit formatting a really long commit that is longer than 100 characters so it should get"`;

exports[`Commit details formatting When there is a type and subject, then they are formatted as the commit header 1`] = `"New: Test commit formatting"`;
